src/backend/utils/activity/pgstat_function.c

```diff
@@ -79,7 +80,7 @@ pgstat_init_function_usage(FunctionCallInfo fcinfo,
 	PgStat_BackendFunctionEntry *pending;
 	bool		created_entry;
 
-	if (pre_function_call_hook)
+	if (pre_function_call_hook && IsTransactionState())
```

Comment:
this change also applies to non babelfish use cases. Is there a use case/test case for this?

Reply:
pre_function_call_hook, this hook type is also added by babel, it's not from community, so it's quite confident to say it won't affect non babelfish use case in this line